src/services/extension-registry-provider.js

```diff
@@ -19,7 +21,41 @@
             reduce = utils.reduce,
             ownKeys = utils.ownKeys,
             toArray = utils.toArray,
-            isFunction = utils.isFunction;
+            isFunction = utils.isFunction,
+            isNaN = utils.isNaN,
+            isString = utils.isString,
+            isNumber = utils.isNumber,
+            isUndefined = utils.isUndefined,
+            // TODO: externalize this validator
+            // so that a custom sorter could be provided
+            invalidVal = -9999,
+            strToUsableNum = function(val) {
+              var test = Number(val);
+              return isNaN(test) ?
+                      invalidVal :
+                      test;
+            },
+            // ensures any provided weight is usable.
+            // see the unit test for weight
+            // - numbers returned as numbers
+            // - undefined returned as -1 (explicit 0 has greater weight)
+            // - strings returned as numbers || -9999 (invalid)
+            // - all other objects returned as -9999 (invalid)
+            // this function is intentionally resilient as it is assumed
+            // extensions could be registered by various parties.  Those
+            // who use sensible weights should be able to trust sorting
+            // placement
+            toWeight = function(value) {
+              return isNaN(value) ?
+                      invalidVal :
+                      isNumber(value) ?
+                        value :
+                        isString(value) ?
+                          strToUsableNum(value) :
+                          isUndefined(value) ?
+                            -1 :
+                            invalidVal;
```

Comment:
I'm OK leaving the behavior undefined if they don't follow the rules.

But I think the last suggestion should always assign the default for anything we can't parse as an integer, which seems reasonable to me and a lot simpler.